src/components/posts/PostsPage.js

```diff
@@ -1,71 +1,154 @@
+import PostForm from './PostForm'
+import client from '../../utils/client'
+import './style.css'
+import { Box, Stack } from '@mui/material'
+import Header from '../Header/Header'
+import dateTimetoRelativeTime from './helperfunctions'
+import { Link } from 'react-router-dom'
 
 const PostsPage = ({ role }) => {
-  const [post, setPost] = useState({ content: '' });
-  const [postResponse, setPostResponse] = useState('');
-  const [posts, setPosts] = useState([]);
+  const [post, setPost] = useState({ content: '' })
+  const [postResponse, setPostResponse] = useState('')
+  const [posts, setPosts] = useState([])
+  const [comment, setComment] = useState('')
+  const [load, setLoad] = useState(true)
+  const [showAllArr, setShowAll] = useState([])
 
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts));
-  }, []);
-
+    client.get('/posts').then((res) => {
+      setPosts(res.data.data.posts)
+    })
+  }, [load])
   const createPost = async (event) => {
-    event.preventDefault();
+    event.preventDefault()
     client
       .post('/post', post)
       .then((res) => {
-        setPostResponse(res.data);
-        setPosts((posts) => [res.data.data.post, ...posts]);
+        setPostResponse(res.data)
+        setPosts((posts) => [res.data.data.post, ...posts])
+        setLoad((x) => !x)
+      })
+      .catch((data) => {
+        console.log(data)
       })
-      .catch((data) => {});
-    setPost(() => ({ content: '' }));
-  };
+    setPost(() => ({ content: '' }))
+  }
 
   const handleChange = (event) => {
-    event.preventDefault();
-    const { value, name } = event.target;
+    event.preventDefault()
+    const { value, name } = event.target
     setPost({
       ...post,
       [name]: value,
-    });
-  };
+    })
+  }
+
+  const createComment = async (event, postId) => {
+    event.preventDefault()
+    client
+      .post(`/post/${postId}/comment`, { comment })
+      .then(() => {
+        setLoad((x) => !x)
+        setComment('')
+      })
+      .catch((data) => {
+        console.log(data)
+      })
+  }
+
+  const addToShowedComments = (id) => {
+    !showAllArr.some((a) => a === id) && setShowAll((x) => [...x, id])
+  }
+
+  const removeFromShowedComments = (id) => {
+    setShowAll((x) => x.filter((c) => c !== id))
+  }
+
+  const handleComment = (event) => {
+    event.preventDefault()
+    const { value } = event.target
+    setComment(value)
+  }
 
   return (
     <>
       <Header role={role} />
       <section className='posts-section'>
         {postResponse.status}
-        <PostForm handleSubmit={createPost} handleChange={handleChange} inputValue={post.content} />
-        <ul className="posts-list">
+        <PostForm
+          handleSubmit={createPost}
+          handleChange={handleChange}
+          inputValue={post.content}
+        />
+        <ul className='posts-list'>
           {posts.map((post, index) => (
             <li key={index} className='post-item'>
```

Comment:
What do you think about potentially extracting some more components here? It seems like quite a lot is happening in this block of JSX..?

Reply:
Agreed! this task will be handled by Elvis and Khateeb, and we meant to do that here as we focused on the functionality and nothing else (like styling and refactoring)